src/Containers/UpgradeView/UpgradeContainer/index.tsx

```diff
@@ -18,12 +18,15 @@ export interface UpgradeContainerProps {
 const UpgradeContainer = ({ name, buildingContext, availableResources }: UpgradeContainerProps) => {
   const { gameState } = useContext(GameSessionState);
   const [loading, onLoading] = useState(false);
+  const [render, setRerender] = useState();
 
   const displayName = stringToTitle(name);
 
+
   const upgradeDate = localStorage.getItem(`${name}_upgrade`);
   const upgradeTime = upgradeDate ? new Date(upgradeDate as string).getTime() - new Date().getTime() : 0;
 
+  useEffect(setRerender as () => void, [buildingContext.level]);
```

Comment:
What is purpose of this useEffect?

Reply:
+1

Reply:
useEffect is performed when building lvl change to setRerender which rerender Fragment with key={render} on it, beacuse conditional render in react only work when we reload fragment when its occur